Algebroids/gap/PathCategories.gi

```diff
@@ -0,0 +1,1039 @@
+    local C, datum, string, labels;
+    
+    C := CapCategory( alpha );
+    
+    datum := MorphismDatum( alpha );
+    
+    if datum[1] = 0 then
+      
+      string := Concatenation( "id(", LaTeXOutput( Source( alpha ) ), ")" );
+      
+    else
+      
+      labels := CollectEntries( LabelsOfMorphisms( CapQuiver( C ) ){MorphismIndices( alpha )} );
+      
+      string := JoinStringsWithSeparator(
+                  ListN( labels,
+                    function ( label )
+                      
+                      if label[2] = 1 then
+                        return label[1];
+                      else
+                        return Concatenation( label[1], "^{", String( label[2] ), "}" );
+                      fi;
+                      
+                    end ), " " );
+      
+    fi;
+    
+    if ValueOption( "OnlyDatum" ) = true then
+      
+      return string;
+      
+    else
+      
+      return Concatenation(
+                "{", LaTeXOutput( Source( alpha ) ), "}-\\left(",
+                "{", string, "}\\right)\\rightarrow",
+                "{", LaTeXOutput( Target( alpha ) ), "}" );
+    
+    fi;
+    
+end );
+
+##
+InstallMethod( AssignSetOfObjects,
+        [ IsPathCategory, IsString ],
+  
+  function ( C, label )
+    local names, func;
+    
+    names := LabelsOfObjects( CapQuiver( C ) );
+    
+    if label = "" and ForAny( names, name -> Int( name ) <> fail ) then
+        Error( "the <label> passed to 'AssignSetOfObjects' must be a non-empty string!\n" );
+    fi;
+    
+    func :=
+      function ( name, o )
+        
+        name := Concatenation( label, ReplacedString( name, "-", "m" ) );
+        MakeReadWriteGlobal( name );
+        DeclareSynonym( name, o );
+        return true;
+        
+    end;
+    
+    ListN( names, SetOfObjects( C ), func );
+    
+end );
+
+##
+InstallOtherMethod( AssignSetOfObjects,
+        [ IsPathCategory ],
+  
+  function ( C )
+    
+    AssignSetOfObjects( C, "" );
+    
+end );
+
+##
+InstallMethod( AssignSetOfGeneratingMorphisms,
+        [ IsPathCategory, IsString ],
+  
+  function ( C, label )
+    local names, morphisms, func;
+    
+    names := LabelsOfMorphisms( CapQuiver( C ) );
+    
+    if label = "" and ForAny( names, name -> Int( name ) <> fail ) then
+        Error( "the <label> passed to 'AssignSetOfGeneratingMorphisms' must be a non-empty string!\n" );
+    fi;
+    
+    morphisms := SetOfGeneratingMorphisms( C );
+    
+    func :=
+      function ( name, m )
+        
+        name := Concatenation( label, ReplacedString( name, "-", "m" ) );
+        MakeReadWriteGlobal( name );
+        DeclareSynonym( name, m );
+        return true;
+        
+    end;
+    
+    ListN( names, morphisms, func );
+    
+end );
+
+##
+InstallOtherMethod( AssignSetOfGeneratingMorphisms,
+        [ IsPathCategory ],
+  
+  function ( C )
+    
+    AssignSetOfGeneratingMorphisms( C, "" );
+    
+end );
+
+##
+InstallOtherMethod( \/,
```

Comment:
Isn't this method already provided by CAP?

Reply:
Yes, only for objects.

Reply:
Yes.

Reply:
Of course, it have removed it.